### ISSUE TITLE
Uninstall not removing attribute

### ISSUE THREAD

**Issue by p24-max** [magento/meta-for-magento2#24]
### Preconditions (*)

1. Version 1.2.0
2. Magento 2.4.5

### Steps to reproduce (*)

1. Install the extension with composer
2. Run setup upgrade
3. (Verify: Attribute google_product_category is created)
4. Uninstall the module: `bin/magento module:uninstall --remove-data --non-composer -v Meta_Catalog` -> Attribute is still remaining
6. Remove the module with `composer remove`
7. Open a product in edit mode in catalog admin, will result in error
```
Type Error occurred when creating object: Meta\Catalog\Model\Config\Source\Product\GoogleProductCategory, Meta\Catalog\Model\Config\Source\Product\GoogleProductCategory::__construct(): Argument #1 ($sourceData) must be of type array, null given, called in ...
```

### Expected result (*)

1. That it can be safely uninstalled

### Actual result (*)

1.  Removing the module causes errors in the magento admin (edit product page):
```
Type Error occurred when creating object: Meta\Catalog\Model\Config\Source\Product\GoogleProductCategory, Meta\Catalog\Model\Config\Source\Product\GoogleProductCategory::__construct(): Argument #1 ($sourceData) must be of type array, null given, called in ...
```

**Comment by willfreeze** [comment]
Got the same issue on Magento 2.4.6
Can I manually remove the google_product_category to resolve the problem?

**Comment by PhiIipp** [comment]
@willfreeze  did you find a way ?

ChatGPT propose the following, and i see the record in the db.

SELECT attribute_id FROM eav_attribute WHERE attribute_code = 'google_product_category';
DELETE FROM catalog_product_entity_* WHERE attribute_id = [attribute_id];
DELETE FROM eav_attribute WHERE attribute_id = [attribute_id];

but i dont want to crash my DB schema , or run in problems when I install a new versions once meta plugin  compatible with magento 2.4.7 and php 8.3

**Comment by mediaweb03** [comment]
I was successful getting rid of this error...
I ran the above recommended mySQL commands:
SELECT attribute_id FROM eav_attribute WHERE attribute_code = 'google_product_category';
It found the attribute_code.

Then I ran...  DELETE FROM eav_attribute WHERE attribute_id = [attribute_id]; 
I did not run the other mysql query since it seemed harder to add back

Logged in my admin panel, and my error was gone.

**Comment by zlik** [comment]
This should be fixed now. Deployed the fix in [v1.3.3](https://github.com/magento/meta-for-magento2/releases/tag/v1.3.3). Please let us know if this becomes an issue again. Thanks!

**Comment by srokatonie** [comment]
I just deleted the attribute and it helped, however, I cannot reproduce the steps.

I believe that you have to make sure that 

`SELECT attribute_id FROM eav_attribute WHERE attribute_code = 'google_product_category';`

Or similar query would return a reference class `Meta\Catalog\Model\Config\Source\Product\GoogleProductCategory`. After I deleted relevant attribute that references the Meta...GoogleProductCategory, the error is gone.
